### PR TITLE
Created functionality for Login Authentiaction

### PR DESCRIPTION
**Pull request by AnaGetovska**
The login authentication logic is structured at AuthContext class and useAuth hook.

### CODE REVIEW

File: src/AuthContext.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+				removeAuth,
+				auth,
+			}}
+		>
+			{children}
+		</AuthContext.Provider>
+	)
+}
+
+export { AuthContext, AuthProvider }
```
Comment (superKalo) [inline src/AuthContext.jsx:38]: Maybe move this to the context(s) directory?

File: src/components/Login.js [hunk 2]
```diff
@@ -1,11 +1,74 @@
...
-        <>
-            <h1><Link to="dashboard">Login</Link></h1>
-        </>
+        <ChakraProvider>
+            <Box bg="purple" paddingY="12em">
+                <form onSubmit={handleSubmit}>
+                    <VStack bg="white" paddingY="5em" marginX="20em" className="form-wrap">
+                        {hasError && <Box>{hasError}</Box>}
+                        <Box>
+                            <Input name="username" placeholder="Потребителско име" required onChange={handleChange} />
```
Comment (superKalo) [inline src/components/Login.js:61]: This should be "факултетен номер", right? Add validation for numbers only.

File: src/components/Login.js [hunk 3]
```diff
@@ -1,11 +1,74 @@
...
+                        </Box>
+                        <Box>
+                            <Input name="password" placeholder="Парола" type="password" required onChange={handleChange} />
```
Comment (superKalo) [inline src/components/Login.js:64]: One more validation rule needed - should be 4 characters or more (as far as I cross-checked on the WebStudent login form)

File: src/components/Login.js [hunk 4]
```diff
@@ -1,11 +1,74 @@
...
+
     return (
-        <>
-            <h1><Link to="dashboard">Login</Link></h1>
-        </>
+        <ChakraProvider>
+            <Box bg="purple" paddingY="12em">
+                <form onSubmit={handleSubmit}>
+                    <VStack bg="white" paddingY="5em" marginX="20em" className="form-wrap">
+                        {hasError && <Box>{hasError}</Box>}
```
Comment (superKalo) [inline src/components/Login.js:59]: Please make the error UI look more like an error :)

File: src/components/Login.js [hunk 5]
```diff
@@ -1,11 +1,74 @@
...
+            faculty_number: formData.username,
+            password: formData.password
+        });
+    }
+
     return (
-        <>
-            <h1><Link to="dashboard">Login</Link></h1>
-        </>
+        <ChakraProvider>
```
Comment (superKalo) [inline src/components/Login.js:55]: Please wrap <ChakraProvider> in the root of the app. Otherwise - no other components will be able to use Chakra components instead this specific one.

File: src/components/Logout.js [hunk 6]
```diff
@@ -1,10 +1,13 @@
...
+import useAuth from '../useAuth'
 
 const Logout = () => {
+    const removeAuth = useAuth().removeAuth;
+    
     return (
         <>
             <h1>Logout</h1>
-            <Link to="../">Login</Link>
+            <Button onClick={removeAuth}>Logout</Button>
```
Comment (superKalo) [inline src/components/Logout.js:10]: When I click it once - it takes me to `/logout` route and nothing happens. When I click it a second time - it logs me out.

File: src/App.js [hunk 7]
```diff
@@ -1,15 +1,22 @@
...
 import Logout from './components/Logout'
+import Login from './components/Login'
+import useAuth from './useAuth'
 
 function App() {
+	const user = useAuth().auth;
+
+	if(!user){
+		return <Login />
+	}
```
Comment (superKalo) [inline src/App.js:14]: This flow works, but it's not optimal and has edge cases. When I log in - the route does not change (from /login to /dashboard for example). When I log out, the route also stays /logout.

Ideally, we need to have [Protecter Routes](https://blog.logrocket.com/complete-guide-authentication-with-react-router-v6/#creating-protected-routes). Then - if someone tries to access /dashboard route and it's not logged in - redirect to /login route.

File: package.json [hunk 8]
```diff
@@ -34,10 +39,11 @@
...
       "last 1 firefox version",
       "last 1 safari version"
-	]
+    ]
   },
   "devDependencies": {
+    "eslint": "^8.7.0",
     "eslint-config-prettier": "^8.6.0",
-    "prettier": "2.8.3"
+    "prettier": "^2.5.1"
```
Comment (superKalo) [inline package.json:47]: Why did you change the prettier lib version?

File: package.json [hunk 9]
```diff
@@ -34,10 +39,11 @@
       "last 1 chrome version",
       "last 1 firefox version",
       "last 1 safari version"
-	]
+    ]
   },
   "devDependencies": {
+    "eslint": "^8.7.0",
```
Comment (superKalo) [inline package.json:45]: Rreate-react-app comes with eslint config. Maybe there is no need to include eslint in our dev dependencies too?

Comment (AnaGetovska) [reply package.json:45]: In our previous conversation you said that maybe I haven't installed eslint prettier. I double-checked and saw in Prettier ESlint prerequisites that I need to install prettier and eslint packages. [ESlint Prettier ReadMe](https://github.com/idahogurl/vs-code-prettier-eslint/blob/master/README.md). Do you think I should remove it?

**Comment by AnaGetovska** [comment]
- Added faculty number validation
- Better error messaging
- Private routing
- Login, Logout, Dashboard redirection added
- Moved useAuth into hooks directory
- Moved AuthContext into context directory

File: package.json (hunk 9 above)
Comment (superKalo) [reply package.json:45]: Yes. These packages (eslint in this case) must be installed either locally or globally. The reate-react-app comes with them as a dep. So no need to manually install them or add them in here. So please remove eslint (`npm uninstall eslint --save-dev`).

File: src/components/Login.js [hunk 10]
```diff
@@ -1,11 +1,123 @@
...
-    )
+	const setAuth = useAuth().setAuth
+	const navigate = useNavigate()
+	const [formData, setFormdata] = useState({
+		facultyNumber: '',
+		password: '',
+	})
+
+	const [hasError, setError] = useState(false)
+	const [isInvalid, setIsInvalid] = useState(false)
```
Comment (superKalo) [inline src/components/Login.js:25]: There is something wrong with the error mechanism - no error gets displayed when I try to log in with wrong credentials.

**Review by superKalo — APPROVED** [review]
